tests/eosio.msig_tests.cpp

```diff
@@ -1265,4 +1245,102 @@ BOOST_FIXTURE_TEST_CASE( assert_if_context_free_actions_in_propose, eosio_msig_t
+                  ("proposal_name", "first")
+                  ("trx",           trx)
+                  ("requested",     vector<permission_level>{{ N(alice), config::active_name },
+                                                             { N(bob), config::active_name }})
+   );
+   BOOST_REQUIRE_EQUAL( false, get_earliest_exec_time(N(alice), N(first)).valid() );
+
+   // cache the correctly expected value of `earliest_exec_time` into `exec_time`.
+   static uint32_t exec_time = calculate_test_exec_time(delay_sec);
+   
+   push_action( N(alice), N(approve), mvo()
+                  ("proposer",      "alice")
+                  ("proposal_name", "first")
+                  ("level",         permission_level{ N(alice), config::active_name })
+   );
+   BOOST_REQUIRE_EQUAL( exec_time, get_earliest_exec_time(N(alice), N(first))->sec_since_epoch() );
+
+   // produce a block to ensure consistency between `earliest_exec_time` and `exec_time`.
+   produce_block();
+
+   push_action( N(bob), N(approve), mvo()
+                  ("proposer",      "alice")
+                  ("proposal_name", "first")
+                  ("level",         permission_level{ N(bob), config::active_name })
+   );
+   BOOST_REQUIRE_EQUAL( exec_time, get_earliest_exec_time(N(alice), N(first))->sec_since_epoch() );
+
+   push_action( N(alice), N(unapprove), mvo()
+                           ("proposer",      "alice")
+                           ("proposal_name", "first")
+                           ("level",         permission_level{ N(alice), config::active_name })
+   );
+   BOOST_REQUIRE_EQUAL( false, get_earliest_exec_time(N(alice), N(first)).valid() );
+
+   // re-cache the correctly expected value of `earliest_exec_time` into `exec_time`.
+   exec_time = calculate_test_exec_time(delay_sec);
+
+   push_action( N(alice), N(approve), mvo()
+                  ("proposer",      "alice")
+                  ("proposal_name", "first")
+                  ("level",         permission_level{ N(alice), config::active_name })
+   );
+   BOOST_REQUIRE_EQUAL( exec_time, get_earliest_exec_time(N(alice), N(first))->sec_since_epoch() );
+
+   // produce a block to ensure consistency between `earliest_exec_time` and `exec_time`.
+   produce_block();
+
+   push_action( N(bob), N(unapprove), mvo()
+                         ("proposer",      "alice")
+                         ("proposal_name", "first")
+                         ("level",         permission_level{ N(bob), config::active_name })
+   );
+   BOOST_REQUIRE_EQUAL( exec_time, get_earliest_exec_time(N(alice), N(first))->sec_since_epoch() );
+   
+   transaction_trace_ptr trx_trace;
+   trx_trace = push_action( N(alice), N(exec), mvo()
+                             ("proposer",      "alice")
+                             ("proposal_name", "first")
+                             ("executer",      "alice")
+   );
+
+   BOOST_REQUIRE( trx_trace->receipt.valid() );
+   BOOST_REQUIRE( bool(trx_trace) );
```

Comment:
Switch order of the above two asserts to avoid undefined behavior:

```cpp
   BOOST_REQUIRE( bool(trx_trace) );
   BOOST_REQUIRE( trx_trace->receipt.valid() );
```

Similarly elsewhere.